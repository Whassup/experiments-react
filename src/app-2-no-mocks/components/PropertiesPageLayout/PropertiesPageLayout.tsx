import { FunctionComponent } from "react";


interface PropertiesPageLayoutProps {
  propertyList: React.ReactNode
}

export const PropertiesPageLayout: FunctionComponent<PropertiesPageLayoutProps> = ({propertyList }) => {
  return <main>
    <h1>Properties Page</h1>
    {propertyList}
  </main>

};
