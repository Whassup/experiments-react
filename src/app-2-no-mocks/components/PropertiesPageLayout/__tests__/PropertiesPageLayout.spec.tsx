import { render, screen } from "@testing-library/react";
import { PropertiesPageLayout } from "../PropertiesPageLayout";


describe("<PropertyCard />", () => {
    const propertyListContent = "Property List"
    const propertyList = <>{propertyListContent}</>;

    describe("render", () => {

        it("should render", () => {
            const {asFragment} = render(<PropertiesPageLayout propertyList={propertyList}/>);
            expect(asFragment()).toMatchSnapshot();
        });
        
        describe("GIVEN propertyList", () => {
            beforeEach(() => {
                render(<PropertiesPageLayout propertyList={propertyList}/>);
            });

            it("SHOULD display property card title", () => {
                const titleElement = screen.getByText(propertyListContent);
                expect(titleElement).toBeInTheDocument();
            });

        });
    });
});