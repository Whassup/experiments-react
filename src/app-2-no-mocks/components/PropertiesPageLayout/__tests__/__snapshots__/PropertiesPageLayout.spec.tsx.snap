// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PropertyCard /> render should render 1`] = `
<DocumentFragment>
  <main>
    <h1>
      Properties Page
    </h1>
    Property List
  </main>
</DocumentFragment>
`;
